File: ClientSide/src/app/pages/studentAttendances/models/studentAttendanceDataModel.ts
import { StudentShortModel } from './studentShortModel';
import { StudentAttendanceGridModel } from './studentAttendanceGridModel';

export class StudentAttendanceDataModel {

    public student: StudentShortModel;
    public studentAttandances: Array<StudentAttendanceGridModel>

    constructor(
        student: StudentShortModel,
        studentAttandances: Array<StudentAttendanceGridModel>
    ) {
        this.student = student;
        this.studentAttandances = studentAttandances;
    }

}

File: ServerSide/SAMS.REST.API/DataModels/StudentAttendanceDataModel.cs
using SAMS.BusinessLogic.Models.StudentAttendances;
using SAMS.BusinessLogic.Models.Students;
using System.Collections.Generic;

namespace SAMS.REST.API.DataModels
{
    public class StudentAttendanceDataModel
    {
        public StudentShortModel Student { get; set; }
        public IEnumerable<StudentAttandanceGridModel> StudentAttandances { get; set; }
    }
}
